go-test-node/env.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	gcrypto "github.com/ethereum/go-ethereum/crypto"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/multiformats/go-multiaddr"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const (
+	myPort = 5000
+	//prometheusPort  = 8008
+	httpPublishPort = 8645
+)
+
+var (
+	inShadow = os.Getenv("SHADOWENV") != ""
+	chunks   = getEnvInt("FRAGMENTS", 1)
+	msgSeen  = make(map[uint64]int)
+	log      = logging.Logger("pubsub-test")
+)
+
+type PublishRequest struct {
+	Topic   string `json:"topic"`
+	MsgSize int    `json:"msgSize"`
+	Version int    `json:"version"`
+}
+
+type PublishResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type PeerInfo struct {
+	PeerName string
+	Addrs    []ma.Multiaddr
+}
+
+func getEnvInt(key string, defaultValue int) int {
+	if value := os.Getenv(key); value != "" {
+		if intVal, err := strconv.Atoi(value); err == nil {
+			return intVal
+		}
+	}
+	return defaultValue
+}
+
+func getHostname() (string, int) {
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		panic(fmt.Errorf("error getting hostname: %s", err))
+	}
+
+	parts := strings.Split(hostname, "-")
+	if len(parts) < 2 {
+		panic(fmt.Errorf("unexpected hostname format: %s", hostname))
+	}
```

Comment:
I don't think we have to enforce the hostname to have this format.
The only important thin I would say is the last `-{n}` part, as in `node-0`, `node-1`...
But what comes before `-0` we don't really care.